tests/test_targets.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+# Copyright (c) 2025 Munich Quantum Software Company GmbH
+# All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+#
+# Licensed under the MIT License
+
+"""Test targets."""
+
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+import pytest
+from qiskit.transpiler import Target
+
+from mqt.bench.targets.devices import get_available_device_names, get_device
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping, Sequence
+
+
+@dataclass(frozen=True)
+class DeviceSpec:
+    """Specification describing an expected device configuration."""
+
+    name: str
+    num_qubits: int
+    single_gates: set[str] = field(default_factory=set)
```

Comment:
```suggestion
    single_qubit_gates: set[str] = field(default_factory=set)
```
Just for consistency.